tools/c7n_azure/c7n_azure/function_package.py

```diff
@@ -228,13 +221,21 @@ def publish(self, deployment_creds):
         # update perms of the package
         self._update_perms_package()
         zip_api_url = '%s/api/zipdeploy?isAsync=true' % deployment_creds.scm_uri
```

Comment:
Why are the user credentials required? I am surprised `deployment_creds.scm_uri` is not sufficient here.